httpbp/middlewares.go

```diff
@@ -246,3 +249,47 @@ func SupportedMethods(method string, additional ...string) Middleware {
 		}
 	}
 }
+
+func PrometheusServerMetrics(serverSlug string, x bool) Middleware {
+	return func(name string, next HandlerFunc) HandlerFunc {
+		return func(ctx context.Context, w http.ResponseWriter, r *http.Request) (err error) {
+			start := time.Now()
+			method := r.Method
+			endpoint := r.URL.Path
+			activeRequestLabels := prometheus.Labels{
+				methodLabel:   method,
+				endpointLabel: endpoint,
+			}
+			serverActiveRequests.With(activeRequestLabels).Inc()
+
+			defer func() {
+				success := strconv.FormatBool(err == nil)
+				code := http.StatusText(http.StatusOK)
+				if err != nil {
+					code = err.Error()
```

Comment:
anyone know a better way to get the status code here that will be used for `http_response_code` label? ref: [baseplate spec for http](https://github.snooguts.net/reddit/baseplate.spec/blob/master/component-apis/prom-metrics.md#http-server-metrics).

Reply:
you'll need to create a ResponseWriter that intercepts it.

```
type codeInterceptor struct {
  http.ResponseWriter
  code int
}

func (ci *codeInterceptor) WriteHeader(code int) {
  ci.code = code
  ci.ResponseWriter.WriteHeader(code)
}
```

Now you make one of those, stick "w" as the "ResponseWriter" field, pass it to next, and check the .code after it returns.

Note that WriteHeader might not be called, so code might be zero, in which case the implied response code is http.StatusOK

Reply:
I just did this with statsd metrics as well:

https://github.com/reddit/baseplate.go/pull/453/files#diff-f0f0f7b278ddd73593a069e51e2d62c8b1fcac7d296258c2e724d85d31c8bb9eR278-R291

Note, you also want to check if the error is an `HTTPError` since returning that will set the status code later in the call (after middeware is done): 

https://github.com/reddit/baseplate.go/pull/453/files#diff-f0f0f7b278ddd73593a069e51e2d62c8b1fcac7d296258c2e724d85d31c8bb9eR359-R362